Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/ViewBookmarkLoader.cpp

```diff
@@ -0,0 +1,552 @@
+                    registry->Remove(bookmarkKey + "/");
+                    return visitedViewBookmarks;
+                }
+            }
+        }
+
+        return false;
+    }
+
+    AZStd::optional<ViewBookmark> ViewBookmarkLoader::GetBookmarkAtIndex(int index, const StorageMode mode)
+    {
+        switch (mode)
+        {
+        case StorageMode::Shared:
+            AZ_Assert(false, "Shared Bookmark Component functionality not implemented.");
+            return AZStd::optional<ViewBookmark>();
+        case StorageMode::Local:
+
+            if (index >= 0 && index < m_localBookmarks.size())
+            {
+                return AZStd::optional<ViewBookmark>(m_localBookmarks.at(index));
+            }
+            AZ_Warning("ViewBookmarkLoader", false, "Couldn't load View Bookmark from file.");
+            return AZStd::optional<ViewBookmark>();
+        case StorageMode::Invalid:
+        default:
+            return AZStd::optional<ViewBookmark>();
+        }
+    }
+
+    bool ViewBookmarkLoader::RemoveBookmarkAtIndex(int index, const StorageMode mode)
+    {
+        switch (mode)
+        {
+        case StorageMode::Shared:
+            AZ_Assert(false, "Shared Bookmark Component functionality not implemented.");
+            return false;
+        case StorageMode::Local:
+            return RemoveLocalBookmarkAtIndex(index);
+        case StorageMode::Invalid:
+        default:
+            AZ_Warning("ViewBookmarkLoader", false, "Couldn't find ViewBookmarkComponent");
+            return false;
+        }
+    }
+
+    AZStd::optional<ViewBookmark> ViewBookmarkLoader::LoadLastKnownLocation() const
+    {
+        return m_lastKnownLocation;
+    }
+
+    LocalViewBookmarkComponent* ViewBookmarkLoader::RetrieveLocalViewBookmarkComponent()
+    {
+        AZ::EntityId levelEntityId;
+        AzToolsFramework::ToolsApplicationRequestBus::BroadcastResult(
+            levelEntityId, &AzToolsFramework::ToolsApplicationRequests::GetCurrentLevelEntityId);
+
+        if (levelEntityId.IsValid())
+        {
+            AZ::Entity* levelEntity = nullptr;
+            AZ::ComponentApplicationBus::BroadcastResult(levelEntity, &AZ::ComponentApplicationBus::Events::FindEntity, levelEntityId);
+            if (levelEntity)
+            {
+                LocalViewBookmarkComponent* bookmarkComponent = levelEntity->FindComponent<LocalViewBookmarkComponent>();
+                if (bookmarkComponent)
+                {
+                    return bookmarkComponent;
+                }
+                //If we didn't find a component then we add it and return it.
+                levelEntity->Deactivate();
+                levelEntity->CreateComponent<LocalViewBookmarkComponent>();
+                levelEntity->Activate();
+                bookmarkComponent = levelEntity->FindComponent<LocalViewBookmarkComponent>();
+
+                if (bookmarkComponent)
```

Comment:
this seems like it could be an assert because it was just created, it should be there.